src/wwwroot/directives/link-toggler.js

```diff
@@ -13,19 +13,16 @@
       var directive = {
         restrict: 'A',
         link: function (scope, element, attrs, ctrls) {
-            var decodedTkn = auth.decodeToken();
-            if(decodedTkn) {
-              var hrefAllowed;
-              if(!attrs.href) {
-                // Finds the first element inside the container and analize.
-                hrefAllowed = element[0].querySelector('a').href.split("#")[1].match(decodedTkn.acceptedUrlsPattern) ? true : false;
-              } else {              
-                hrefAllowed = attrs.href.match(decodedTkn.acceptedUrlsPattern) ? true : false;
-              }
-              if(!hrefAllowed) {
-                element.remove();
-              }
-            }
+          var hrefAllowed;
+          if(!attrs.href) {
+            // Finds the first element inside the container and analize.
+            hrefAllowed = auth.isUrlAllowed(element[0].querySelector('a').href.split("#")[1]);
```

Comment:
Did you test it with multiples `a` elements?

Reply:
yes, it will only analize the first one. (and remove or not the container if the first "A" should be removed)

Reply:
Talking with @mvillaseco, he explain me that _querySelector_ returns the first element that matches the query.